### ISSUE TITLE
Feature Request: Marked messages.

### ISSUE THREAD

**Issue by dotancohen** [peter-iakovlev/Telegram#262]
Telegram should support the ability to Mark / Star / Highlight / Label individual messages for quick reference in the future. Most other chat and messaging software, such as Whatsapp and Slack, provide this feature.

This feature request is forked from #25 (Starred Messages). That issue has been closed because the OP felt that the Saved Messages feature fulfilled his needs. Here I list some (not all) differences between Marked Messages and the current Saved Messages feature:

1. The Star is not available in the context of the thread.
2. Often I will need to remember many messages, such as a list of tasks sent in multiple messages. The current implementation requires me to forward each message, instead of simply going to the first starred message to find the rest. The link to the original message feature is not helpful here as I may not know if additional messages are to be addressed out of context.
3. The forwarding requires a context switch, which precludes quickly validating or rejecting messaged tasks.
4. The saved messages for all contacts are all in a single "metacontact". This is inappropriate if I am at a client's site and I have messages open from other clients or from personal contacts.

In short, the Saved Messages feature is great for single, actionable items. However, some messages that users want to reference are not actionable, but rather bits that one might like to reference some time in the future. Maybe a sweet expression of love, or a cute picture. I don't need to be reminded that these messages exist, but I want to be able to find them quickly if ever I need to remember that my daughter loves me or I want to see my son's first steps again.

**Comment by dotancohen** [comment]
Since issue #25 has been reopened, I am closing this issue.
